pkg/apis/environment/extension.go

```diff
@@ -150,3 +153,25 @@ func getCaps(entities model.Services) (int, int) {
 
 	return verticesCap, edgesCap
 }
+
+func (h Handler) RouteCloneEnvironment(req RouteCloneEnvironmentRequest) (*RouteCloneEnvironmentResponse, error) {
+	entity := req.Model()
+
+	for i := range req.Variables {
+		v := entity.Edges.Variables[i]
+		if v == nil {
+			return nil, errorx.New("invalid input: nil variable")
+		}
+
+		if v.Sensitive && v.Value == "" {
+			v.Value = h.modelClient.Variables().Query().
```

Comment:
Can we query vars once instead of querying one for each var?

Reply:
updated